ee/query-service/app/server.go

```diff
@@ -199,17 +202,61 @@ func (s *Server) createPrivateServer(apiHandler *api.APIHandler) (*http.Server,
 	}, nil
 }
 
+func getPATToken(r *http.Request) (string, error) {
+	patHeader := r.Header.Get("SIGNOZ-API-KEY")
+	if patHeader == "" {
+		return "", nil
+	}
+
+	authHeaderParts := strings.Fields(patHeader)
+	if len(authHeaderParts) != 2 || strings.ToLower(authHeaderParts[0]) != "bearer" {
+		return "", fmt.Errorf("PAT authorization header format must be bearer {token}")
+	}
+
+	return authHeaderParts[1], nil
+}
+
 func (s *Server) createPublicServer(apiHandler *api.APIHandler) (*http.Server, error) {
 
 	r := mux.NewRouter()
 
+	getUserFromPAT := func(r *http.Request) (*model.UserPayload, error) {
+		patToken, err := getPATToken(r)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get PAT token in request headers, err: %v", err)
+		}
+		ctx := context.Background()
+		dao := apiHandler.AppDao()
+		pat, err := dao.GetPAT(ctx, patToken)
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch PAT token from DB, err %v", err)
+		}
+		user, apierr := dao.GetUser(ctx, pat.UserID)
+		if apierr != nil {
+			return nil, fmt.Errorf("failed to fetch user for PAT from DB, err: %v", apierr)
+		}
+		return user, nil
+	}
+
+	getUserFromRequest := func(r *http.Request) (*model.UserPayload, error) {
+		user, err := getUserFromPAT(r)
+		if err == nil && user != nil {
+			zap.S().Debugf("Found valid PAT user: %+v", user)
+			return user, nil
+		}
+		if err != nil {
+			zap.S().Debugf("Error while getting user for PAT: %+v", err)
+		}
+		return baseauth.GetUserFromRequest(r)
+	}
+	am := baseapp.NewAuthMiddleware(getUserFromRequest)
 	r.Use(setTimeoutMiddleware)
 	r.Use(s.analyticsMiddleware)
 	r.Use(loggingMiddleware)
 
-	apiHandler.RegisterRoutes(r)
-	apiHandler.RegisterMetricsRoutes(r)
-	apiHandler.RegisterLogsRoutes(r)
+	apiHandler.RegisterRoutes(r, am)
+	apiHandler.RegisterMetricsRoutes(r, am)
+	apiHandler.RegisterLogsRoutes(r, am)
 
 	c := cors.New(cors.Options{
```

Comment:
My understanding is cors is usually for browsers. But do we need to add "SIGNOZ-API-KEY" in the allowed headers. 
May we have to omit it to disable api calls from browser.  I have no clarity on this. But do give it a thought

Reply:
Based on my understanding, we'll need to add it to allowed headers. CORS is not just for browsers but for any request coming from other origin. Thanks for pointing it out.